scylla/src/transport/topology.rs

```diff
@@ -251,6 +255,9 @@ impl MetadataReader {
         let mut result = self.fetch_metadata(initial).await;
         if let Ok(metadata) = result {
             self.update_known_peers(&metadata);
+            if initial {
+                self.check_that_control_connection_is_an_accepted_host(&metadata);
```

Comment:
Would it be possible to do the same filtering that we do in `update_knoiwn_peers` in the constructor?
We would be sure that the control connection is connected to an accepted peer.

Forcing the user to specify only known_peers that are accepted by the host filter seems overly restrictive.
I can imagine that someone might want to write a single app with predefined list of peers and then deploy it in multiple DCs, something like:
```rust
let session = SessionBuilder::new()
    .known_peers(["eu-1", "eu-2", "eu-3", "us-1", "us-2", "us-3"])
    .host_filter(make_host_filter_for_dc(get_dc_from_env_variable()));
    .build()
```

Reply:
It would be nice if it were possible, but unfortunately not all kinds of filters can be implemented this way. For example, we don't really know which nodes are in which DC until we fetch topology data from system.local/system.peers.